libraries/botbuilder-lg/src/activityFactory.ts

```diff
@@ -0,0 +1,361 @@
+import { MessageFactory, CardFactory } from 'botbuilder-core';
+
+export class ActivityFactory {
+    private static GenericCardTypeMapping: Map<string, string> = new Map<string, string>
+    ([
+        [ 'herocard', CardFactory.contentTypes.heroCard ],
+        [ 'thumbnailcard', CardFactory.contentTypes.thumbnailCard ],
+        [ 'audiocard', CardFactory.contentTypes.audioCard ],
+        [ 'videocard', CardFactory.contentTypes.videoCard ],
+        [ 'animationcard', CardFactory.contentTypes.animationCard ],
+        [ 'signincard', CardFactory.contentTypes.signinCard ],
+        [ 'oauthcard', CardFactory.contentTypes.oauthCard ],
+    ]);
+
+    public static CreateActivity(lgStringResult: string): Activity {
+        let lgStructuredResult: any;
+        try {
+            lgStructuredResult = JSON.parse(lgStringResult);
+        } catch (error) {
+            return this.buildActivityFromText(lgStructuredResult.trim());
+        }
+
+        return this.buildActivityFromLGStructuredResult(lgStructuredResult);
+    }
+
+    private static buildActivityFromText(text: string): Activity {
+        return MessageFactory.text(text, text) as Activity;
+    }
+
+    private static buildActivityFromLGStructuredResult(lgJObj: any): Activity {
+        let activity: any = {};
+
+        const type: string = this.getStructureType(lgJObj);
+        if (ActivityFactory.GenericCardTypeMapping.has(type)) {
+            const attachment: Attachment = this.getAttachment(lgJObj);
+            if (attachment !== undefined) {
+                activity = MessageFactory.attachment(attachment);
+            }
+        } else if (type === 'activity') {
+            activity = this.buildActivityFromObject(lgJObj);
+        } else {
+            throw new Error(`type ${ type } is not support currently.`);
+        }
+
+        return activity;
+    }
+
+    private static buildActivityFromObject(lgJObj: any): Activity {
+        let activity: any = {};
+
+        if ('type' in lgJObj && lgJObj.type === 'event') {
+            activity = this.buildEventActivity(lgJObj);
+        } else {
+            activity = this.buildMessageActivity(lgJObj);
+        }
+
+        return activity;
+    }
+
+    private static buildEventActivity(lgJObj: any): Activity {
+        let activity: any = { type: ActivityTypes.Event };
+        for (const item of Object.keys(lgJObj)) {
+            const property: string = item.trim();
+            const value: any = lgJObj[item];
+            switch (property.toLowerCase()) {
+                case '$type':
+                    break;
+                case 'name':
+                    activity.name = value.toString();
+                    break;
+                case 'value':
+                    activity.value = value.toString();
+                    break;
+                default:
+                    console.log(`Skipping unknown activity property ${ property }`);
```

Comment:
There are potentially other valid property types. Do we really want to restrict the properties they can add to the activity?